tests/PHPStan/Analyser/data/param-out.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace ParamOut;
+
+use function PHPStan\Testing\assertType;
+
+/**
+ * @param-out string $s
+ */
+function addFoo(?string &$s): void
+{
+	if ($s === null) {
+		$s = "hello";
+	}
+	$s .= "foo";
+}
+
+/**
+ * @template T of int
+ * @param-out T $s
+ */
+function genericFoo(mixed &$s): void
+{
+}
```

Comment:
played a bit with the psalm implementtion and found this interessting generic + param-out cases: https://psalm.dev/r/f30310f1f6